rskj-core/src/main/java/co/rsk/peg/PeginInformation.java

```diff
@@ -0,0 +1,102 @@
+import co.rsk.peg.pegininstructions.PeginInstructionsVersion1;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PeginInformation {
+
+    private static final Logger logger = LoggerFactory.getLogger(PeginInformation.class);
+
+    private final BtcLockSenderProvider btcLockSenderProvider;
+    private final PeginInstructionsProvider peginInstructionsProvider;
+
+    private int protocolVersion;
+    private RskAddress rskDestinationAddress;
+    private Address btcRefundAddress;
+
+    public PeginInformation(
+        BtcLockSenderProvider btcLockSenderProvider,
+        PeginInstructionsProvider peginInstructionsProvider) {
+        this.btcLockSenderProvider = btcLockSenderProvider;
+        this.peginInstructionsProvider = peginInstructionsProvider;
+    }
+
+    public int getProtocolVersion() {
+        return protocolVersion;
+    }
+
+    public RskAddress getRskDestinationAddress() {
+        return this.rskDestinationAddress;
+    }
+
+    public Address getBtcRefundAddress() {
+        return this.btcRefundAddress;
+    }
+
+    public void parse(BtcTransaction btcTx) throws PeginInstructionsException {
+        logger.debug("[parse] Trying to parse peg-in information from btc tx {}", btcTx.getHash());
+
+        // Get information from tx sender first
+        Optional<BtcLockSender> btcLockSenderOptional = btcLockSenderProvider.tryGetBtcLockSender(btcTx);
+        if(btcLockSenderOptional.isPresent()) {
+            BtcLockSender btcLockSender = btcLockSenderOptional.get();
+            parseFromBtcLockSender(btcLockSender);
+        }
+
+        // If peg-in instructions were provided then override the info obtained from BtcLockSender
+        Optional<PeginInstructions> peginInstructionsOptional = peginInstructionsProvider.buildPeginInstructions(btcTx);
+        if (peginInstructionsOptional.isPresent()) {
+            PeginInstructions peginInstructions = peginInstructionsOptional.get();
+            parseFromPeginInstructions(peginInstructions);
+        }
+
+        // If BtcLockSender could not be parsed and peg-in instructions were not provided, then this tx can't be processed
+        if(!btcLockSenderOptional.isPresent() && !peginInstructionsOptional.isPresent()) {
+            String message = String.format("Could not get peg-in information for tx %s", btcTx.getHash());
+            logger.error("[parse] {}", message);
+            throw new PeginInstructionsException(message);
+        }
+    }
+
+    private void parseFromBtcLockSender(BtcLockSender btcLockSender) {
+        this.protocolVersion = 0;
+        this.rskDestinationAddress = btcLockSender.getRskAddress();
+        this.btcRefundAddress = btcLockSender.getBTCAddress();
+
+        logger.info("[parseFromBtcLockSender] RSK destination address: {}", btcLockSender.getRskAddress());
+        logger.info("[parseFromBtcLockSender] BTC refund address: {}", btcLockSender.getBTCAddress().toBase58());
```

Comment:
```suggestion
        logger.trace("[parseFromBtcLockSender] RSK destination address: {}", btcLockSender.getRskAddress());
        logger.trace("[parseFromBtcLockSender] BTC refund address: {}", btcLockSender.getBTCAddress());
```
Besides the logging level, you don't need to call `toBase58`. Internally this method calls `toString` which is automatically called by the log, if the logging level is enabled.